Week2/todo_app/program.js

```diff
@@ -22,51 +22,151 @@ class TodoModel {
     }
 
     create(description, callback) {
-        // Write code and query to create a new TODO item
+        const createQuery = `
+        INSERT INTO todo_items(text , is_completed , user_id)
+        VALUES (${dbConnection.escape(description)} , 0 ,${dbConnection.escape(args2)} )`;
+        this.dbConnection.query(createQuery, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            }
+            callback(null, results);
+        });
     }
 
     update(id, description, callback) {
-        // Write code and query to update and existing TODO item
+        const updateQuery = `
+        UPDATE todo_items SET text = ${dbConnection.escape(description)} 
+        WHERE id = ${dbConnection.escape(id)}`;
+        this.dbConnection.query(updateQuery, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            }
+            callback(null, results);
+        });
     }
 
     delete(id, callback) {
-        // Write code and query to delete an existing TODO item
+        const deleteQuery = `
+        DELETE FROM todo_items
+        WHERE id = ${dbConnection.escape(id)}`;
+        this.dbConnection.query(deleteQuery, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            }
+            callback(null, results);
+        });
     }
 
     tagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query add a tag to a TODO item
+        const tagInsertQuery = `
+        INSERT INTO todo_item_tag(todo_item_id , tag_id) 
+        VALUES (${dbConnection.escape(todoItemId)} , ${dbConnection.escape(tagId)})`;
+        this.dbConnection.query(tagInsertQuery, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            }
+            callback(null, results);
+        });
     }
         
     untagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query remove a tag from a TODO item
+        const tagDeleteQuery = `
+        DELETE FROM todo_item_tag
+        WHERE todoItemId = ${dbConnection.escape(todoItemId)} AND tagId = ${dbConnection.escape(tagId)}`;
+        this.dbConnection.query(tagDeleteQuery, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            }
+            callback(null, results);
+        });
     }
 
     markCompleted(todoItemId, callback) {
-        // Write code to mark a TODO item as completed
+        const updateQuery = `
+        UPDATE todo_items SET is_completed = 1
+        WHERE ${dbConnection.escape(todoItemId)} = id`;
+        this.dbConnection.query(updateQuery, (err, results, fields) => {
```

Comment:
:+1: 

Keep in mind `is_completed` is a bool, not an int - so `TRUE` would be a better value to set.